.github/workflows/pr-greeter.yml

```diff
@@ -0,0 +1,27 @@
+name: PR Greeter
+
+on:
+  pull_request_target:
+   types: [ opened ]
+
+permissions:
+  pull-requests: write
```

Comment:
The convention we've been using for permissions is to set `contents: read` for the whole file and then write permissions like this are added to the job.  See https://github.com/llvm/llvm-project/blob/main/.github/workflows/release-binaries.yml for example.